File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem30_OddEvenLinkedListTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;
import org.phoenix.leetcode.challenges.Problem29_DeleteTheMiddleNodeOfALinkedList.ListNode;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem30_OddEvenLinkedListTest {

    private final Problem30_OddEvenLinkedList test = new Problem30_OddEvenLinkedList();

    @Test
    void oddEvenList() {
        ListNode head = new ListNode(1);
        head.next = new ListNode(2);
        head.next.next = new ListNode(3);
        head.next.next.next = new ListNode(4);
        head.next.next.next.next = new ListNode(5);

        assertEquals(3, test.oddEvenList(head).next.val);
    }
}

File: JavaScript/leetcode75/src/test/Problem30_OddEvenLinkedList.test.js
const oddEvenList = require("../main/Problem30_OddEvenLinkedList");

function ListNode(val, next) {
  this.val = val === undefined ? 0 : val;
  this.next = next === undefined ? null : next;
}

test("should return odd list before even", () => {
  let list = new ListNode(1, 2);
  list.next = new ListNode(2, 3);
  list.next.next = new ListNode(3, 4);
  list.next.next.next = new ListNode(4, 5);
  expect(oddEvenList(list).next.val).toBe(3);
});
